src/main/java/com/google/research/bleth/simulator/Simulation.java

```diff
@@ -1,9 +1,380 @@
 package com.google.research.bleth.simulator;
 
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * An abstract class representing a BLETH simulation.
+ * Can be either a Tracing simulation or a Stalking simulation.
+ */
 public abstract class Simulation {
-    Board board;
 
-    public Board getBoard() {
+    private final String id;
+    protected int currentRound = 0;
+    private final int maxNumberOfRounds;
+    private Board board;
+    protected ArrayList<Beacon> beacons;
```

Comment:
If we don't change beacons during the Simulation, prefer to use ImmutableList to store them.